pkg/ebpf/extension.go

```diff
@@ -0,0 +1,83 @@
+package ebpf
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// Extension interface for Tracee
+type Extension interface {
+	// Initialize the extension
+	Init(t *Tracee) error
+	// Register initialization events which are emitted by the extension
+	RegisterInitEvents(t *Tracee, out chan *trace.Event) error
+	// Run the extension
+	Run(ctx context.Context, t *Tracee) error
+	// Close the extension
+	Close() error
+}
+
+type ExtensionCallbacks struct {
```

Comment:
If ExtensionCallbacks is for Events consider naming it as EventExtensionCallbacks.